nodes/analyzer/weighted_vector_sum.py

```diff
@@ -0,0 +1,453 @@
+    bl_idname = 'SvWeightedVectorSumNode'
+    bl_label = 'Weighted Vector Sum (Alpha)'
+    bl_icon = 'SNAP_FACE_CENTER'
+    sv_icon = 'SV_CENTROID'
+
+
+    def updateSizeSocket(self, context):
+        if self.center_mode == 'VERTICES':
+            self.outputs['output_sizes'].label = 'Count of verts'
+            self.outputs['output_size' ].label = 'Total Verts'
+        elif self.center_mode == 'EDGES':
+            self.outputs['output_sizes'].label = 'Lengths'
+            self.outputs['output_size' ].label = 'Total Length'
+        elif self.center_mode == 'FACES':
+            self.outputs['output_sizes'].label = 'Areas'
+            self.outputs['output_size' ].label = 'Total Area'
+        elif self.center_mode == 'VOLUMES':
+            self.outputs['output_sizes'].label = 'Volumes'
+            self.outputs['output_size' ].label = 'Total Volume'
+        
+        updateNode(self, context)
+
+    center_modes = [
+        ("VERTICES", "Vertices", "Calc center of mass by vertices and ignore edges, faces and volumes", 0),
+        ("EDGES", "Edges", "Calc center of mass by length of edges and ignore vertices, faces and volumes", 1),
+        ("FACES", "Faces", "Calc center of mass by surfaces of objects and ignore vertices, edges and volumes", 2),
+        ("VOLUMES", "Volumes", "Calc center of mass by volume of objects and ignore vertices, edges and faces", 3)
+    ]
+
+    quad_modes = [
+        ("BEAUTY", "Beauty", "Split the quads in nice triangles, slower method", 1),
+        ("FIXED", "Fixed", "Split the quads on the 1st and 3rd vertices", 2),
+        ("ALTERNATE", "Fixed Alternate", "Split the quads on the 2nd and 4th vertices", 3),
+        ("SHORT_EDGE", "Shortest Diagonal", "Split the quads based on the distance between the vertices", 4)
+    ]
+
+    ngon_modes = [
+        ("BEAUTY", "Beauty", "Arrange the new triangles nicely, slower method", 1),
+        ("EAR_CLIP", "Clip", "Split the ngons using a scanfill algorithm", 2)
+    ]
+
+    skip_unmanifold_centers: BoolProperty(
+        name='Skip unmanifold centers',
+        description='If True then skip unmanifold centers else show Exception\n\nExample:\n   True: if object has only vertices and \'Center mode\'=\'Volumes\', \'Faces\' or \'Edges\' then skip this object without exception',
+        default=True,
+        update=updateNode) # type: ignore
+
+    center_mode: EnumProperty(
+        name='Center mode',
+        description="Calc center mode",
+        items=center_modes,
+        default="VOLUMES",
+        update=updateSizeSocket) # type: ignore
+
+    quad_mode: EnumProperty(
+        name='Quads mode',
+        description="Quads processing mode",
+        items=quad_modes,
+        default="BEAUTY",
+        update=updateNode) # type: ignore
+
+    ngon_mode: EnumProperty(
+        name="Polygons mode",
+        description="Polygons processing mode",
+        items=ngon_modes,
+        default="BEAUTY",
+        update=updateNode) # type: ignore
```

Comment:
I'd suggest to add a checkbox to disable triangulation at all (maybe hide it to the N panel). In case the user knows that the mesh is already tris-only.

Reply:
If mode is 'Face' then one can check all faces are tris without any property:
![image](https://github.com/nortikin/sverchok/assets/14288520/fe042d8c-2f71-4551-a184-ae9638c27132)

but for mode 'Volume' it is not enough: mesh has to be closed and this is a difficult operation. I have suggestion for property 'skip_test_volume_are_closed' and this do real economy  for volume node:

![image](https://github.com/nortikin/sverchok/assets/14288520/c218c98f-b77a-4d6e-86ad-1b78c0eb3419)

Reply:
But this mode do triangulation if some faces are not tris and this back to slowdown performance but user get good result.